nestkernel/node_collection.cpp

```diff
@@ -294,6 +294,14 @@ NodeCollectionPrimitive::to_array() const
 
 NodeCollectionPTR NodeCollectionPrimitive::operator+( NodeCollectionPTR rhs ) const
 {
+  if ( rhs->empty() )
+  {
+    return NodeCollectionPTR( new NodeCollectionPrimitive( *this ) );
+  }
+  if ( empty() )
+  {
+    return NodeCollectionPTR( rhs );
```

Comment:
Is this consistent with all other cases? In my understanding, `operator+()` should always return a new object, and does so in all other cases, but here we just get a new pointer to an existing object. This might lead to subtle problems later. I would suggest
```suggestion
    return NodeCollectionPTR( new NodeCollectionPrimitive( *rhs ) );
```